modules/xfem/src/userobjects/LevelSetCutUserObject.C

```diff
@@ -0,0 +1,157 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#include "LevelSetCutUserObject.h"
+#include "SubProblem.h"
+#include "MooseVariable.h"
+#include "libmesh/string_to_enum.h"
+
+template <>
+InputParameters
+validParams<LevelSetCutUserObject>()
+{
+  InputParameters params = validParams<GeometricCutUserObject>();
+  params.addRequiredParam<VariableName>(
+      "level_set_var", "The name of level set variable used to represent the interface");
+  params.addClassDescription("XFEM mesh cut by level set function");
+  return params;
+}
+
+LevelSetCutUserObject::LevelSetCutUserObject(const InputParameters & parameters)
+  : GeometricCutUserObject(parameters),
+    _ls_var_number(
+        _subproblem.getVariable(_tid, parameters.get<VariableName>("level_set_var")).number()),
+    _system(_subproblem.getSystem(getParam<VariableName>("level_set_var"))),
+    _solution(_system.current_local_solution.get())
+{
+}
+
+bool
+LevelSetCutUserObject::cutElementByGeometry(const Elem * elem,
+                                            std::vector<Xfem::CutEdge> & cut_edges,
+                                            std::vector<Xfem::CutNode> & /*cut_nodes*/,
+                                            Real /*time*/) const
+{
+  bool cut_elem = false;
+
+  unsigned int n_sides = elem->n_sides();
+
+  for (unsigned int i = 0; i < n_sides; ++i)
+  {
+    UniquePtr<Elem> curr_side = elem->side(i);
+
+    if (curr_side->type() != EDGE2)
```

Comment:
Probably should add a note in the doxygen comments and the forthcoming *.md files about this limitation on the types of element edges

Reply:
Maybe it works for higher order edge as well. Let me check.

Reply:
@jiangwen84 Have you checked on that?  I'm OK merging the code with this limitation, but we should get it working for higher order elements.

Reply:
Yes, I checked it. For second order elements, the side type is still EDGE2. So this check is fine. I added a regression test in the second_order_elemnts folder.